runtime/lua/vim/lsp/handlers.lua

```diff
@@ -53,7 +53,8 @@ local function progress_handler(_, result, ctx, _)
       end
     end
   else
-    table.insert(client.messages, {content = val, show_once = true, shown = 0})
+    client.messages.progress[token] = val
+    client.messages.progress[token].done = true
```

Comment:
I am not sure if this is always safe. Theoretically `val` could be a string, number of boolean and then this would fail.